### ISSUE TITLE
[Bug]: Cycle detected when a Datasource references a JSObject returning a Table widget 

### ISSUE THREAD

**Issue by osangenis** [appsmithorg/appsmith#20659]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

I can bind  a GET API request `Binance` header or parameter to a table property like `{{ Table1.sortOrder.column }}`. However if I declare a JS object `JSObject1` like 
```
export default {
	getTable: () => {
		return Table1
	}
}
```
and bind the parameter like `{{JSObject1.getTable().sortOrder.column }}` an error appears: 
```
Cyclical dependency error encountered while parsing relationship [Binance.data : Table1.tableData] where the relationship is denoted as (source : target).
```

There are workarounds but the code could be simpler if we could use JS methods in API query parameters, particularly useful for building complex query strings.

### Steps To Reproduce

1. Create any API request (example: `https://data.binance.com/api/v3/ticker/24hr`)
2. Create a Table `Table1` 
3. Bind `Table1` to the API Request
4. Create a JS object with the function described above
5. Add one parameter to the API Request and set the value as  {{JSObject1.getTable().sortOrder.column }}

### Version

appsmith-ce:v1.8.12

**Comment by bharath31** [comment]
@osangenis I agree that JSObject functions are useful for building complex query strings. Returning a cyclic reference to the Table object could be a bad idea. I would suggest running the API using a param instead. Let me know if this works: 

**JSObject:** 
```
export default {
    runApi: (keyword, order) => {
        let queryString = `sortColumn=${Table1.sortOrder.column}&sortOrder=${order}&limit=50&term=${keyword}`
        Api1.run({query: queryString})
    }
}
```


**API:**
`https://data.binance.com/api/v3/ticker/24hr?query={{this.params.query}}`

**Comment by Nikhil-Nandagopal** [comment]
@bharath31 shouldn't this work since without this pagination wouldn't be possible?

**Comment by bharath31** [comment]
@Druthi could you please run an RCA here?

**Comment by Nikhil-Nandagopal** [comment]
@morandalex @abeatbeyondlab we'll look into this but the occurrence of this seems fairly rare. You can easily get around fixing this by changing the way you are accessing the code and we'll be happy to suggest changes to your code to unblock you.
